src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/NuGetWorkspaceCommandHandler.cs

```diff
@@ -65,24 +79,64 @@ public bool QueryStatus(List<WorkspaceVisualNodeBase> selection, Guid pguidCmdGr
                     {
                         case PkgCmdIDList.CmdidRestorePackages:
-                            var isRestoreActionInProgress = _restoreCommandHandler.IsRestoreActionInProgress();
-                            cmdf = (uint)((isRestoreActionInProgress ? 0 : Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED)
-                                | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED);
-                            handled = true;
+                            if (IsSolutionOnlySelection(selection))
+                            {
+
+                                var isRestoreActionInProgress = _restoreCommandHandler.IsRestoreActionInProgress();
+                                cmdf = (uint)((isRestoreActionInProgress ? 0 : Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED)
+                                    | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED);
+                                handled = true;
+                            }
+                            break;
+                        case PkgCmdIDList.CmdIdManageProjectUI:
+                            if (IsSupportedProjectSelection(selection))
+                            {
+                                var isPackageManagerUISupported = _packageManagerUICommandHandler.IsPackageManagerUISupported(selection.Single());
+                                cmdf = (uint)(isPackageManagerUISupported ?
+                                    (Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED) :
+                                    Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_INVISIBLE);
+                                handled = true;
+                            }
+                            break;
+                        default:
                             break;
                     }
                 }
             }
-
             return handled;
         }
 
+        private static bool IsSupportedProjectSelection(List<WorkspaceVisualNodeBase> selection)
+        {
+            if (selection != null &&
+                selection.Count.Equals(1))
+            {
+                // We support every item representing a project
+                // - we don't have the ability to do a capabilities check, because we don't have enough information.
+                string fileExtension;
+                try
+                {
+                    fileExtension = Path.GetExtension(selection.Single().NodeMoniker);
+                }
+                catch (ArgumentException)
+                {
+                    return false;
+                }
+
+                if (fileExtension == null)
+                {
+                    return false;
+                }
+                // We do not know if the project is supported
+                return fileExtension.EndsWith("proj", StringComparison.OrdinalIgnoreCase);
```

Comment:
Do we have a list of supported project types or is it calculated via `VsHierarchy.IsCapabilityMatch()`? Just like [here](https://github.com/NuGet/NuGet.Client/blob/c2466e779f42bf308a232531f11cb47d7ba99de2/src/NuGet.Clients/NuGet.VisualStudio.Common/VsHierarchyUtility.cs#L118)

Reply:
We'd normally call: https://github.com/NuGet/NuGet.Client/blob/c2466e779f42bf308a232531f11cb47d7ba99de2/src/NuGet.Clients/NuGet.VisualStudio.Common/VsHierarchyUtility.cs#L64, but those types are not available.

Reply:
Can we create an issue for traceability to fix these type of occurrences once the required types are available?

Reply:
We can, but there's so many things missing right now. 
Tracking all of those would be an overkill. I prefer to track larger experiences and rely on feedback for the smaller ones.

Happy to chat if you disagree.